connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Loggers.java

```diff
@@ -0,0 +1,213 @@
+        Map<String, LoggerLevel> result = new TreeMap<>();
+
+        Enumeration<org.apache.log4j.Logger> enumeration = currentLoggers();
+        Collections.list(enumeration)
+                .stream()
+                .filter(logger -> logger.getLevel() != null)
+                .forEach(logger -> result.put(logger.getName(), loggerLevel(logger)));
+
+        org.apache.log4j.Logger root = rootLogger();
+        if (root.getLevel() != null) {
+            result.put(ROOT_LOGGER_NAME, loggerLevel(root));
+        }
+
+        return result;
+    }
+
+    /**
+     * Set the level for the specified logger and all of its children
+     * @param namespace the name of the logger to adjust along with its children; may not be nul
+     * @param level the level to set for the logger and its children; may be null
+     * @return all loggers that were affected by this action; may be empty, but never null
+     */
+    public synchronized List<String> setLevel(String namespace, Level level) {
+        Objects.requireNonNull(namespace, "Logging namespace may not be null");
+
+        log.info("Setting level of namespace {} and children to {}", namespace, level);
+        List<org.apache.log4j.Logger> childLoggers = loggers(namespace);
+
+        List<String> result = new ArrayList<>();
+        for (org.apache.log4j.Logger logger: childLoggers) {
+            setLevel(logger, level);
+            result.add(logger.getName());
+        }
+        Collections.sort(result);
+
+        return result;
+    }
+
+    /**
+     * Retrieve all known loggers within a given namespace, creating an ancestor logger for that
+     * namespace if one does not already exist
+     * @param namespace the namespace that the loggers should fall under; may not be null
+     * @return all loggers that fall under the given namespace; never null, and will always contain
+     * at least one logger (the ancestor logger for the namespace)
+     */
+    private synchronized List<org.apache.log4j.Logger> loggers(String namespace) {
+        Objects.requireNonNull(namespace, "Logging namespace may not be null");
+
+        if (ROOT_LOGGER_NAME.equalsIgnoreCase(namespace)) {
+            List<org.apache.log4j.Logger> result = Collections.list(currentLoggers());
+            result.add(rootLogger());
+            return result;
+        }
+
+        List<org.apache.log4j.Logger> result = new ArrayList<>();
+        org.apache.log4j.Logger ancestorLogger = lookupLogger(namespace);
+        Enumeration<org.apache.log4j.Logger> en = currentLoggers();
+        boolean present = false;
+        while (en.hasMoreElements()) {
+            org.apache.log4j.Logger current = en.nextElement();
+            if (current.getName().startsWith(namespace)) {
+                result.add(current);
+            }
+            if (namespace.equals(current.getName())) {
+                present = true;
+            }
+        }
+
+        if (!present) {
+            result.add(ancestorLogger);
+        }
```

Comment:
> I'm hesitant to tweak anything here that may end up causing a regression in the existing worker-scoped API

Yeah, that's fair enough. This PR is pretty large as it is and it probably makes more sense to minimize changes to these core areas being refactored to avoid regressions 👍